crates/optimism/txpool/src/validator.rs

```diff
@@ -364,3 +337,46 @@ impl OpForkTracker {
         self.interop.load(Ordering::Relaxed)
     }
 }
+
+/// Extracts commitment from access list entries, pointing to 0x420..022 and validates them
+/// against supervisor.
+///
+/// If commitment present pre-interop tx rejected.
+///
+/// Returns:
+/// None - if tx is not cross chain,
+/// Some(Ok(()) - if tx is valid cross chain,
+/// Some(Err(e)) - if tx is not valid or interop is not active
+pub async fn is_valid_cross_tx(
+    access_list: Option<&AccessList>,
+    hash: &TxHash,
+    timestamp: u64,
+    timeout: Option<u64>,
+    is_interop_active: bool,
+    client: Option<&SupervisorClient>,
+) -> Option<Result<(), InvalidCrossTx>> {
```

Comment:
this function could also be moved into the supervisorclient directly, but we can do this separately as well

Reply:
done